clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +112,125 @@ def version(self) -> str:
         return self._server_conf.version
 
 
+def _extract_region_from_endpoint(endpoint):
+    """
+    Extract the region name from an STS endpoint URL.
+    for example: https://sts.eu-central-1.amazonaws.com/ -> eu-central-1
+    and for example: https://sts.amazonaws.com/ -> DEFAULT_REGION
+
+    :param endpoint: The endpoint URL of the STS client.
+    :return: The region name extracted from the endpoint URL.
+    """
+
+    parts = endpoint.split('.')
+    if len(parts) == 4:
+        return parts[1]
+    if len(parts) > 4:
+        return parts[2]
+    return DEFAULT_REGION
+
+
+def _get_identity_token(
+        session: 'boto3.Session',
+        lakefs_host: str,
+        additional_headers: dict[str, str] = None,
+        presign_expression=60
+) -> str:
+    """
+   Generate the identity token required for lakeFS authentication from an AWS session.
+
+   This function uses the STS client to generate a presigned URL for the `get_caller_identity` action,
+    extracts the required values from the URL,
+   and creates a base64-encoded JSON object with these values.
+
+   :param session: A boto3 session object with the necessary AWS credentials and region information.
+   :return: A base64-encoded JSON string containing the required authentication information.
+   :raises ValueError: If the session does not have a region name set.
+   """
+
+    # this method should only be called when installing the aws-iam additional requirement
+    from botocore.client import Config  # pylint: disable=import-outside-toplevel, import-error
+    from botocore.signers import RequestSigner  # pylint: disable=import-outside-toplevel, import-error
+
+    sts_client = session.client('sts', config=Config(signature_version='v4'))
+    endpoint = sts_client.meta.endpoint_url
+    service_id = sts_client.meta.service_model.service_id
+    region = _extract_region_from_endpoint(endpoint)
+    # signer is used because the presigned URL generated by the STS does not support additional headers
+    signer = RequestSigner(
+        service_id,
+        region,
+        'sts',
+        'v4',
+        session.get_credentials(),
+        session.events
+    )
+    endpoint_with_params = f"{endpoint}/?Action=GetCallerIdentity&Version=2011-06-15"
+    if additional_headers is None:
+        additional_headers = {
+            'X-LakeFS-Server-ID': lakefs_host,
+        }
+    params = {
+        'method': 'POST',
+        'url': endpoint_with_params,
+        'body': {},
+        'headers': additional_headers,
+        'context': {}
+    }
+
+    presigned_url = signer.generate_presigned_url(
+        params,
+        region_name=region,
+        expires_in=presign_expression,
+        operation_name=''
+    )
+    parsed_url = urlparse(presigned_url)
+    query_params = parse_qs(parsed_url.query)
+
+    # Extract values from query parameters
+    print(query_params)
```

Comment:
use logger to DEBUG instead or remove.  
I would prefer to **keep it** when debugging that's ULTRA helpful.

Reply:
oops, removed